modules/cimg/CMakeLists.txt

```diff
@@ -62,57 +66,60 @@ target_compile_definitions(inviwo-module-cimg PRIVATE
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Iex
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IexMath
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Imath
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IlmThread
-        ${CMAKE_CURRENT_BINARY_DIR}/ext/openexr/openexr/OpenEXR/config
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImf
-        ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImfUtil
-    )
+option(IVW_USE_OPENEXR  "Add OpenEXR image format support (HDR images)" ON)
+option(IVW_USE_EXTERNAL_OPENEXR  "OpenEXR is provided externaly" OFF)
+if(IVW_USE_OPENEXR)
     target_compile_definitions(inviwo-module-cimg PRIVATE cimg_use_openexr)
+    if(IVW_USE_EXTERNAL_OPENEXR)
+        set(CMAKE_FIND_PACKAGE_PREFER_CONFIG OFF)
+        find_package(OpenEXR REQUIRED)
+        target_link_libraries(inviwo-module-cimg PRIVATE ${OpenEXR_LIBRARIES})
+        set(extraIncDir ${OpenEXR_INCLUDE_DIRS})
+        list(TRANSFORM extraIncDir APPEND "/OpenEXR") # CImg uses includes without OpendEXR
+        target_include_directories(inviwo-module-cimg PRIVATE ${OpenEXR_INCLUDE_DIRS} ${extraIncDir})
+        ivw_vcpkg_install(openexr MODULE CImg)
+        # OpenEXR from vcpkg is missing this define
+        target_compile_definitions(inviwo-module-cimg PRIVATE OPENEXR_DLL)
+    else()
+        add_subdirectory(ext/openexr)
+        ivw_register_license_file(NAME "OpenEXR" TARGET OpenEXR::IlmImf MODULE CImg
+            URL "http://www.openexr.com"
+            TYPE "Modified BSD License"
+            FILES ext/openexr/openexr/LICENSE.md
+        )
+        ivw_default_install_targets(
+            Half
+            Iex
+            IexMath
+            Imath
+            IlmThread
+            IlmImf
+            IlmImfUtil
+        )
+        target_link_libraries(inviwo-module-cimg
+            PRIVATE
+                IlmBase::Half
+                IlmBase::Iex
+                IlmBase::IexMath
+                IlmBase::Imath
+                IlmBase::IlmThread
+                OpenEXR::IlmImf
+                OpenEXR::IlmImfUtil
+        )
+        target_include_directories(inviwo-module-cimg PRIVATE 
+            ${CMAKE_CURRENT_BINARY_DIR}/ext/openexr/openexr/IlmBase/config
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Half
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Iex
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IexMath
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/Imath
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/IlmBase/IlmThread
+            ${CMAKE_CURRENT_BINARY_DIR}/ext/openexr/openexr/OpenEXR/config
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImf
+            ${CMAKE_CURRENT_SOURCE_DIR}/ext/openexr/openexr/OpenEXR/IlmImfUtil
+        )
```

Comment:
Vcpkg thingy?

Reply:
was there before too. OpenEXR does not like to be a submodule...